### PR TITLE
Regression in msg_generator when escaping trailing double-quotes

### PR DESCRIPTION
**Pull request by dhood**
Does it make sense to add tests for genpy to check each of the std_msgs?

### CODE REVIEW

**Comment by dhood** [comment]
although, even Empty for me is giving full_text='\n'. But something else might show up (in light of https://github.com/ros/genpy/commit/4f7755b3863755db8ed6f8e8549ae26e9f650526#commitcomment-16880685)

File: src/genpy/generator.py [hunk 1]
```diff
@@ -772,7 +772,7 @@ def msg_generator(msg_context, spec, search_path):
     
     full_text = compute_full_text_escaped(msg_context, spec)
     # escape trailing double-quote, unless already escaped, before wrapping in """
-    if full_text[-1] == '"' and not full_text[-2:] == r'\"':
+    if len(full_text) > 0 and full_text[-1] == '"' and not full_text[-2:] == r'\"':
```
Comment (dirk-thomas) [inline src/genpy/generator.py]: I thought about something similar yesterday :wink: but only tried it with a string of length 1 which the list slicing if obviously fine with...

Anyway I think we should make it even more simple and defensive now:

    if full_text.endswith('"') and not full_text.endswith(r'\"'):

**Comment by dirk-thomas** [comment]
Adding checks for every std_msgs seems to be extensive. Especially since `genpy` can't depend on `std_msgs` and therefore would need to duplicate all message definitions. More test cases would always be good but in this case I think it's not strictly necessary. Fixing this logic problem should be "enough".

File: src/genpy/generator.py (hunk 1 above)
Comment (dhood) [inline src/genpy/generator.py]: Ah, I hadn't seen this before! will use it from now on, thanks

**Comment by dhood** [comment]
ok - here's hoping :)

**Comment by dirk-thomas** [comment]
Great. Thanks!

**Pull request merged**